backend/test_observer/data_access/models.py

```diff
@@ -207,6 +208,28 @@ def is_kernel(self) -> bool:
         """Kernel artefacts start with 'linix-' or end with '-kernel'"""
         return self.name.startswith("linux-") or self.name.endswith("-kernel")
 
+    def _get_latest_builds(self) -> list["ArtefactBuild"]:
+        # Group builds by architecture
+        grouped_builds = defaultdict(list)
+        for build in self.builds:
+            grouped_builds[build.architecture].append(build)
+
+        return [
+            max(builds, key=lambda b: b.revision if b.revision else 0)
```

Comment:
@omar-selo can you please help me verify that it is fine to assign revision to 0 if it is None. Mypy was complaining because the key here can be null.

Reply:
I think this should be fine